src/browser.ts

```diff
@@ -1,41 +1,95 @@
 import { History, Location, Action } from 'history';
 import createBrowserHistory from 'history/createBrowserHistory';
 import trim = require('lodash/trim');
-import omit = require('lodash/omit');
 import * as qs from 'qs';
+import find = require('lodash/find');
+import mapValues = require('lodash/mapValues');
+import omit = require('lodash/omit');
+import difference = require('lodash/difference');
 
-// TODO: below, ignoring path params for the moment
-// there's only `view` which is the whole `pathname` from history perspective
+export type PathConfig<S extends BrowserState> = {
+  deserialize: (pathname: string) => Partial<S> | null;
+  serialize: (state: Partial<S>) => string;
+  is: (state: S) => boolean;
+  pick: (state: S) => Partial<S>;
+};
+export type PathConfigs<S extends BrowserState> = Array<PathConfig<S>>;
 
 export type BrowserState = { [k: string]: string } & { view: string };
 
+const defaultNoPath: PathConfig<BrowserState> = {
+  deserialize: (pathname: string) => ({ view: pathname }),
+  serialize: ({ view: pathname }: BrowserState) => pathname,
+  is: () => true,
+  pick: s => ({ view: s.view })
+};
+
+function pathnameToPathConfig(pathname: string, paths: PathConfigs<BrowserState>): PathConfig<BrowserState> {
+  return find(paths, p => Boolean(p.deserialize(pathname))) || defaultNoPath;
+}
+
+function locationToPathname(location?: Location): string {
+  return location ? trim(location.pathname, ' /') : '';
+}
+
+function locationToPathConfig(paths: PathConfigs<BrowserState>, location?: Location): PathConfig<BrowserState> {
+  return pathnameToPathConfig(locationToPathname(location), paths);
+}
+
+function parsePathname(paths: PathConfigs<BrowserState>, location?: Location): Partial<BrowserState> {
+  return locationToPathConfig(paths, location).deserialize(locationToPathname(location))!;
+}
+
 // export is for tests only
 // TODO(gio): `location` here is optional just because (I think)
 // it can be undefined in some tests using `memoryHistory`
-export function parseBrowserState(location?: Location): BrowserState {
-  const view = location ? trim(location.pathname, ' /') : '';
+export function parseBrowserState(paths: PathConfigs<BrowserState>, location?: Location): BrowserState {
+  const pathState = parsePathname(paths, location);
   const s = location ? qs.parse(trim(location.search, ' ?')) : {};
-  return { ...s, view };
+  return { ...s, ...pathState };
+}
+
+function cleanupPathParams(
```

Comment:
can you describe me the purpose of this function?

Reply:
answered below, basically "cleanup" path params for the old path but leaving there path params for the new path